codersdk/richparameters.go

```diff
@@ -145,12 +145,18 @@ func validationEnabled(param TemplateVersionParameter) bool {
 // correctly validates.
 // @typescript-ignore ParameterResolver
 type ParameterResolver struct {
-	Rich []WorkspaceBuildParameter
+	Rich       []WorkspaceBuildParameter
+	Transition WorkspaceTransition
 }
 
 // ValidateResolve checks the provided value, v, against the parameter, p, and the previous build.  If v is nil, it also
 // resolves the correct value.  It returns the value of the parameter, if valid, and an error if invalid.
 func (r *ParameterResolver) ValidateResolve(p TemplateVersionParameter, v *WorkspaceBuildParameter) (value string, err error) {
+	// Assume value for required, ephemeral parameter for stopped/deleted workspace
+	if p.Ephemeral && r.Transition != "" && r.Transition != WorkspaceTransitionStart {
```

Comment:
In what situation will we get an empty transition?

Reply:
I guess, only unit tests?